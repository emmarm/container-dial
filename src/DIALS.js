const DIALS = [
  {
    siteName: 'Udemy',
    siteUrl: 'www.udemy.com',
    container: 'Personal',
    favicon: 'https://www.udemy.com/staticx/udemy/images/v6/favicon-196x196.png'
  },
  {
    siteName: 'Youtube',
    siteUrl: 'www.youtube.com',
    container: 'Personal',
    favicon: 'https://s.ytimg.com/yts/img/favicon_96-vflW9Ec0w.png'
  },
  {
    siteName: 'Youtube',
    siteUrl: 'www.youtube.com',
    container: 'Default',
    favicon: 'https://s.ytimg.com/yts/img/favicon_96-vflW9Ec0w.png'
  },
  {
    siteName: '23 Snaps',
    siteUrl: 'www.23snaps.com',
    container: 'Personal',
    favicon: 'https://www.23snaps.com/favicon.ico'
  },
  {
    siteName: 'Shut Up and Sit Down',
    siteUrl: 'www.shutupandsitdown.com',
    container: 'Personal',
    favicon:
      'https://www.shutupandsitdown.com/wp-content/uploads/2016/04/cropped-favicon-200x200.png'
  },
  {
    siteName: 'Facebook',
    siteUrl: 'www.facebook.com',
    container: 'Personal',
    favicon: 'https://static.xx.fbcdn.net/rsrc.php/yo/r/iRmz9lCMBD2.ico'
  },
  {
    siteName: 'Firebase',
    siteUrl: 'firebase.google.com',
    container: 'Personal',
    favicon:
      'https://firebase.google.com/_static/be157a52c2/images/firebase/touchicon-180.png'
  },
  {
    siteName: 'Daily Cooking Quest',
    siteUrl: 'www.dailycookingquest.com',
    container: 'Default',
    favicon: 'https://s.ytimg.com/yts/img/favicon_96-vflW9Ec0w.png'
  },
  {
    siteName: 'Sean Croxton',
    siteUrl: 'www.seancroxton.com/',
    container: 'Personal',
    favicon: 'http://seancroxton.com/apple-touch-icon.png'
  },
  {
    siteName: 'Mozilla Developer Documentation',
    siteUrl: 'developer.mozilla.org',
    container: 'Default',
    favicon: 'https://s.ytimg.com/yts/img/favicon_96-vflW9Ec0w.png'
  }
];

export default DIALS;
