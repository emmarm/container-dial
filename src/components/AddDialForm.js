import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as actions from '../actions';
import getFavicon from '../utils/getFavicon';

class AddDialForm extends Component {
  state = {
    siteName: '',
    siteUrl: ''
  };

  onSiteNameChange = e => {
    const siteName = e.target.value;
    this.setState(() => ({ siteName }));
  };

  onSiteUrlChange = e => {
    const siteUrl = e.target.value;
    this.setState(() => ({ siteUrl }));
  };

  handleSubmit = async e => {
    e.preventDefault();
    const { siteName, siteUrl } = this.state;
    const { container, addDial, handleHideAddDialModal } = this.props;
    const favicon = await getFavicon(siteUrl);
    const dial = {
      siteName,
      siteUrl,
      container: container.name,
      favicon
    };

    addDial(dial);
    handleHideAddDialModal(e);
  };

  render() {
    return (
      <form className="add-dial-modal__form">
        <div className="add-dial-modal__field">
          <label htmlFor="site-name" className="field__label">
            Site Name
          </label>
          <input
            id="site-name"
            className="field__input"
            type="text"
            placeholder="e.g. Facebook"
            onChange={this.onSiteNameChange}
            value={this.state.siteName}
          />
        </div>

        <div className="add-dial-modal__field">
          <label htmlFor="site-url" className="field__label">
            Site URL
          </label>
          <input
            id="site-url"
            className="field__input"
            type="text"
            placeholder="e.g. https://facebook.com"
            onChange={this.onSiteUrlChange}
            value={this.state.siteUrl}
          />
        </div>

        <div className="add-dial-modal__buttons">
          <button
            onClick={this.props.handleHideAddDialModal}
            className="add-dial-modal__button button--secondary"
          >
            Cancel
          </button>

          <button
            className="add-dial-modal__button button--primary"
            onClick={this.handleSubmit}
            style={{ backgroundColor: this.props.theme.primary }}
          >
            Add Dial
          </button>
        </div>
      </form>
    );
  }
}

AddDialForm.propTypes = {
  handleHideAddDialModal: PropTypes.func.isRequired,
  theme: PropTypes.objectOf(PropTypes.string)
};

const mapStateToProps = state => ({
  container: state.container
});

export default connect(
  mapStateToProps,
  actions
)(AddDialForm);
