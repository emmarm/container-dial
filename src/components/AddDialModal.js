import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';

const AddDialModal = ({ handleHideAddDialModal, container, isOpen, theme }) => (
  <Modal
    isOpen={isOpen}
    onRequestClose={handleHideAddDialModal}
    closeTimeoutMS={150}
    className="add-dial-modal"
  >
    <h2 className="add-dial-modal__title">
      New{' '}
      <span className="title__span" style={{ color: theme.primary }}>
        {container.name}
      </span>{' '}
      Dial
    </h2>

    <form className="add-dial-modal__form">
      <div className="add-dial-modal__field">
        <label htmlFor="site-name" className="field__label">
          Site Name
        </label>
        <input
          id="site-name"
          className="field__input"
          type="text"
          placeholder="e.g. Facebook"
        />
      </div>

      <div className="add-dial-modal__field">
        <label htmlFor="site-url" className="field__label">
          Site URL
        </label>
        <input
          id="site-url"
          className="field__input"
          type="text"
          placeholder="e.g. https://facebook.com"
        />
      </div>

      <div className="add-dial-modal__buttons">
        <button
          onClick={handleHideAddDialModal}
          className="add-dial-modal__button button--secondary"
        >
          Cancel
        </button>

        <button
          className="add-dial-modal__button button--primary"
          style={{ backgroundColor: theme.primary }}
        >
          Add Dial
        </button>
      </div>
    </form>
  </Modal>
);

AddDialModal.propTypes = {
  handleHideAddDialModal: PropTypes.func.isRequired,
  isOpen: PropTypes.bool.isRequired,
  container: PropTypes.objectOf(PropTypes.string)
};

export default AddDialModal;
