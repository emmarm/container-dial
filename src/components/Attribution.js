import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

const Text = styled('p')({
  color: 'white',
  fontSize: 14,
  fontStyle: 'italic',
  fontWeight: 100,
  marginRight: 40,
  textAlign: 'right'
});

const Link = styled('a')(
  {
    textDecoration: 'none'
  },
  ({ theme }) => ({
    color: theme.dark
  })
);

const Attribution = ({ image }) => {
  const { user } = !!image && image;
  return (
    <Text>
      Photo by{' '}
      <Link
        href={`${user &&
          user.links.html}?utm_source=container_dial&utm_medium=referral`}
      >
        {user && user.first_name} {user && user.last_name}
      </Link>{' '}
      on{' '}
      <Link href="https://unsplash.com/?utm_source=container_dial&utm_medium=referral">
        Unsplash
      </Link>
    </Text>
  );
};

Attribution.propTypes = {
  image: PropTypes.object
};

export default Attribution;
