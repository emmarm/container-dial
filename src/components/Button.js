import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

import Icon from './Icon';

const Btn = styled('button')(
  {
    alignItems: 'center',
    border: 'none',
    borderRadius: 4,
    boxShadow:
      'rgba(50, 50, 93, 0.1) 0px 7px 14px, rgba(0, 0, 0, 0.08) 0px 3px 6px',
    color: 'white',
    cursor: 'pointer',
    display: 'flex',
    fontSize: 20,
    fontWeight: 100,
    height: 50,
    justifyContent: 'center',
    padding: '0 25px',
    ':disabled': {
      backgroundColor: '#777',
      cursor: 'auto'
    }
  },
  ({ danger, narrow, padded, primary, theme }) => ({
    backgroundColor: primary ? theme.primary : danger ? '#FF3762' : '#bbb',
    padding: narrow && '0 15px',
    paddingLeft: padded && 30,
    ':hover': {
      backgroundColor: primary ? theme.light : danger ? '#FF5F81' : '#c8c8c8',
      ':disabled': {
        backgroundColor: '#777'
      }
    }
  })
);

const Button = ({
  className,
  danger,
  disabled,
  icon,
  narrow,
  onClick,
  primary,
  submit,
  text
}) => (
  <Btn
    className={className}
    danger={danger}
    disabled={disabled}
    onClick={onClick}
    narrow={narrow}
    padded={!!icon && !!text}
    primary={primary}
    type={submit ? 'submit' : 'button'}
  >
    {text}
    {icon && <Icon icon={icon} padded={!!text} />}
  </Btn>
);

Button.propTypes = {
  className: PropTypes.string,
  danger: PropTypes.bool,
  disabled: PropTypes.bool,
  icon: PropTypes.string,
  onClick: PropTypes.func,
  primary: PropTypes.bool,
  submit: PropTypes.bool,
  text: PropTypes.string
};

export default Button;
