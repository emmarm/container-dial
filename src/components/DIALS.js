const DIALS = [
  {
    name: 'Udemy',
    url: 'www.udemy.com',
    container: 'Personal',
    favicon: 'https://www.udemy.com/staticx/udemy/images/v6/favicon-196x196.png'
  },
  {
    name: 'Youtube',
    url: 'www.youtube.com',
    container: 'Personal',
    favicon: 'https://s.ytimg.com/yts/img/favicon_96-vflW9Ec0w.png'
  },
  {
    name: 'Youtube',
    url: 'www.youtube.com',
    container: 'Default',
    favicon: 'https://s.ytimg.com/yts/img/favicon_96-vflW9Ec0w.png'
  },
  {
    name: '23 Snaps',
    url: 'www.23snaps.com',
    container: 'Personal',
    favicon: 'https://www.23snaps.com/favicon.ico'
  },
  {
    name: 'Shut Up and Sit Down',
    url: 'www.shutupandsitdown.com',
    container: 'Personal',
    favicon:
      'https://www.shutupandsitdown.com/wp-content/uploads/2016/04/cropped-favicon-200x200.png'
  },
  {
    name: 'Facebook',
    url: 'www.facebook.com',
    container: 'Personal',
    favicon: 'https://static.xx.fbcdn.net/rsrc.php/yo/r/iRmz9lCMBD2.ico'
  },
  {
    name: 'Firebase',
    url: 'firebase.google.com',
    container: 'Personal',
    favicon:
      'https://firebase.google.com/_static/be157a52c2/images/firebase/touchicon-180.png'
  },
  {
    name: 'Daily Cooking Quest',
    url: 'www.dailycookingquest.com',
    container: 'Default'
  },
  {
    name: 'Sean Croxton',
    url: 'www.seancroxton.com/',
    container: 'Personal',
    favicon: 'http://seancroxton.com/apple-touch-icon.png'
  },
  {
    name: 'Mozilla Developer Documentation',
    url: 'developer.mozilla.org',
    container: 'Default'
  }
];

export default DIALS;
