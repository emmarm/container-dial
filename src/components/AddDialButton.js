import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

import Button from './Button';

const AddButton = styled(Button)({
  background: 'rgba(255, 255, 255, 0.3)',
  borderRadius: 0,
  height: '100%',
  ':hover': {
    background: 'rgba(255, 255, 255, 0.5)'
  }
});

const AddDialButton = ({ handleShowDialModal }) => (
  <AddButton onClick={handleShowDialModal} text="Add New" icon="add" />
);

AddDialButton.propTypes = {
  handleShowDialModal: PropTypes.func.isRequired
};

export default AddDialButton;
