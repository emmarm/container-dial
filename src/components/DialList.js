import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { css } from 'emotion';
import styled from 'react-emotion';
import Sortable from 'react-sortablejs';

import * as actions from '../actions';
import Dial from './Dial';
import EditDialButton from './EditDialButton';
import AddDialButton from './AddDialButton';

const sortable = css({
  display: 'grid',
  gridAutoRows: 80,
  gridGap: 25,
  gridTemplateColumns: 'repeat(auto-fill, 240px)',
  height: '80vh',
  justifyContent: 'center',
  overflowX: 'hidden',
  overflowY: 'auto',
  padding: '10px 40px',
  width: '100vw'
});

const ghost = css({
  opacity: 0
});

const DialContainer = styled('div')({
  display: 'flex',
  position: 'relative',
  ':hover': {
    '& .edit-button': {
      opacity: 1
    },
    '& .handle': {
      opacity: 1
    }
  }
});

const Handle = styled('div')(
  {
    cursor: 'move',
    height: 80,
    opacity: 0,
    position: 'absolute',
    transition: 'all 200ms ease-in',
    width: 20
  },
  ({ theme }) => ({
    ':hover': {
      background: theme.light
    }
  })
);

export class DialList extends Component {
  state = {
    dials: []
  };

  componentDidUpdate(prevProps) {
    const { dials } = this.props;
    if (dials !== prevProps.dials) {
      this.setState(() => ({ dials }));
    }
  }

  renderDials = () =>
    this.state.dials.sort((a, b) => a.sortIndex > b.sortIndex).map(dial => (
      <DialContainer key={dial.id} data-id={dial.id}>
        <Handle className="handle" />
        <Dial ariaLabel={dial.siteName} dial={dial} />
        <EditDialButton
          dial={dial}
          handleShowDialModal={this.props.handleShowDialModal}
        />
      </DialContainer>
    ));

  render() {
    return (
      <Sortable
        className={sortable}
        onChange={order => {
          const dialsOnly = order.filter(item => !!Number(item));
          const newDialOrder = dialsOnly.map((id, index) => {
            return { id, sortIndex: index };
          });

          this.props.updateDialOrder(newDialOrder);
        }}
        options={{
          ghostClass: ghost,
          handle: '.handle',
          onMove: evt => {
            if (evt.related.type === 'button') {
              return false;
            }
          }
        }}
      >
        {this.renderDials()}
        <AddDialButton handleShowDialModal={this.props.handleShowDialModal} />
      </Sortable>
    );
  }
}

DialList.propTypes = {
  dials: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleShowDialModal: PropTypes.func.isRequired,
  updateDialOrder: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  dials: state.dials.filter(
    dial => dial.container === state.container.cookieStoreId
  )
});

export default connect(
  mapStateToProps,
  actions
)(DialList);
