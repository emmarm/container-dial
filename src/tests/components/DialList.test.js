import React from 'react';
import { shallow } from 'enzyme';

import { DialList } from '../../components/DialList';
import DIALS from '../seedData/DIALS';

describe('DialList component', () => {
  const state = { dials: DIALS.filter(dial => dial.container === '123') };
  const props = {
    dials: DIALS.filter(dial => dial.container === '123'),
    handleShowDialModal: () => {},
    updateDialOrder: () => {}
  };
  const wrapper = shallow(<DialList {...props} />);
  wrapper.setState(state);

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders AddDialButton component', () => {
    const addDialButton = wrapper.find('AddDialButton');
    expect(addDialButton.length).toBe(1);
  });

  it('renders a Dial and EditDialButton component for each dial item', () => {
    const dialComponents = wrapper.find('Dial');
    const editDialButtons = wrapper.find('Connect(EditDialButton)');
    expect(dialComponents.length).toBe(3);
    expect(editDialButtons.length).toBe(3);
  });
});
