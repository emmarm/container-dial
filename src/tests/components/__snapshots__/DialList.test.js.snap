// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialList component matches snapshot 1`] = `
<Sortable
  className="css-ikjpvo"
  onChange={[Function]}
  options={
    Object {
      "ghostClass": "css-q7lffx",
      "handle": ".handle",
      "onMove": [Function],
    }
  }
  style={Object {}}
  tag="div"
>
  <Styled(div)
    data-id={1}
    key="1"
  >
    <Styled(div)
      className="handle"
    />
    <Dial
      ariaLabel="Udemy"
      dial={
        Object {
          "container": "123",
          "favicon": "https://www.udemy.com/staticx/udemy/images/v6/favicon-196x196.png",
          "id": 1,
          "siteName": "Udemy",
          "siteUrl": "www.udemy.com",
          "sortIndex": 0,
        }
      }
    />
    <Connect(EditDialButton)
      dial={
        Object {
          "container": "123",
          "favicon": "https://www.udemy.com/staticx/udemy/images/v6/favicon-196x196.png",
          "id": 1,
          "siteName": "Udemy",
          "siteUrl": "www.udemy.com",
          "sortIndex": 0,
        }
      }
      handleShowDialModal={[Function]}
    />
  </Styled(div)>
  <Styled(div)
    data-id={2}
    key="2"
  >
    <Styled(div)
      className="handle"
    />
    <Dial
      ariaLabel="Youtube"
      dial={
        Object {
          "container": "123",
          "favicon": "https://s.ytimg.com/yts/img/favicon_96-vflW9Ec0w.png",
          "id": 2,
          "siteName": "Youtube",
          "siteUrl": "www.youtube.com",
          "sortIndex": 1,
        }
      }
    />
    <Connect(EditDialButton)
      dial={
        Object {
          "container": "123",
          "favicon": "https://s.ytimg.com/yts/img/favicon_96-vflW9Ec0w.png",
          "id": 2,
          "siteName": "Youtube",
          "siteUrl": "www.youtube.com",
          "sortIndex": 1,
        }
      }
      handleShowDialModal={[Function]}
    />
  </Styled(div)>
  <Styled(div)
    data-id={4}
    key="4"
  >
    <Styled(div)
      className="handle"
    />
    <Dial
      ariaLabel="Facebook"
      dial={
        Object {
          "container": "123",
          "favicon": "https://static.xx.fbcdn.net/rsrc.php/yo/r/iRmz9lCMBD2.ico",
          "id": 4,
          "siteName": "Facebook",
          "siteUrl": "www.facebook.com",
          "sortIndex": 2,
        }
      }
    />
    <Connect(EditDialButton)
      dial={
        Object {
          "container": "123",
          "favicon": "https://static.xx.fbcdn.net/rsrc.php/yo/r/iRmz9lCMBD2.ico",
          "id": 4,
          "siteName": "Facebook",
          "siteUrl": "www.facebook.com",
          "sortIndex": 2,
        }
      }
      handleShowDialModal={[Function]}
    />
  </Styled(div)>
  <AddDialButton
    handleShowDialModal={[Function]}
  />
</Sortable>
`;
